lib/psabpf_table.c

```diff
@@ -55,6 +55,89 @@ void psabpf_table_entry_ctx_free(psabpf_table_entry_ctx_t *ctx)
     close_object_fd(&(ctx->prefixes.fd));
     close_object_fd(&(ctx->tuple_map.fd));
     close_object_fd(&(ctx->cache.fd));
+
+    if (ctx->direct_counters_ctx != NULL) {
+        for (unsigned i = 0; i < ctx->n_direct_counters; i++) {
+            psabpf_direct_counter_ctx_free(&ctx->direct_counters_ctx[i]);
+        }
+        free(ctx->direct_counters_ctx);
+        ctx->direct_counters_ctx = NULL;
+    }
+    ctx->n_direct_counters = 0;
+}
+
+enum direct_object_command {
+    DIRECT_OBJECT_COUNT,
+    DIRECT_OBJECT_INIT_CTX,
+};
+
+static int execute_direct_objects_command(psabpf_table_entry_ctx_t *ctx, enum direct_object_command command)
```

Comment:
I'm not sure if `command` terminology is ok here. I would call this operation adding... I don't see for now good choice...

Reply:
Refactored